test/JLM_Tests/src/org/openj9/test/java/lang/management/TestOpenJ9DiagnosticsMXBean.java

```diff
@@ -0,0 +1,564 @@
+				File dumpFile = new File(fileName);
+
+				if (!(agent.equals("stack"))) {
+					found = findAndDeleteFile(dir, dumpFile.getName());
+					Assert.assertTrue(found, fileName + " not found");
+					
+				} 
+			
+			} catch (java.lang.IllegalArgumentException e) {
+				/* stack agent is not supported by the trigger method */
+				if (agent.equals("stack")) {
+					Assert.assertTrue(true, "Did not receive the expected exception IllegalArgumentException");
+				} 
+				else {
+					Assert.fail("Unexpected IllegalArgumentException : " + e.getMessage());
+				}
+			} catch (Exception e) {
+				Assert.fail("Unexpected Exception thrown : " + e.getMessage());
+				e.printStackTrace();
+			}
+
+			index++;
+		}
+	}
+
+	/**
+	 * Function to test if a class heap dump is created
+	 *
+	 * @param diagBean - OpenJ9DiagnosticsMXBean instance that has already been initialized
+	 */	
+	private static void test_triggerClassicHeapDump(OpenJ9DiagnosticsMXBean diagBean) {
+		boolean found = false;
+
+		try {
+			String fileName = diagBean.triggerClassicHeapDump();
+			found = findAndDeleteFile(".", new File(fileName).getName());
+			Assert.assertTrue(found, fileName + " not found");
+		} catch (InvalidDumpOptionException e){
+			Assert.fail("Unexpected InvalidDumpOptionException : " + e.getMessage());
+		} catch (Exception e){
+			Assert.fail("Unexpected Exception thrown : " + e.getMessage());
+			e.printStackTrace();
+		}
+
+	}	
+
+
+	/**
+	 * Function to test if the dump options were reset to what it was at JVM initialization
+	 *
+	 * @param diagBean - OpenJ9DiagnosticsMXBean instance that has already been initialized
+	 */	
+	private void test_resetDumpOptions(OpenJ9DiagnosticsMXBean diagBean) {
+		boolean found = false;
+		try {
+			diagBean.setDumpOptions("java+heap+system:events=vmstop");			
+		} catch (InvalidDumpOptionException e) {
+			Assert.fail("Unexpected InvalidDumpOptionException : " + e.getMessage());
+		} catch (Exception e) {
+			Assert.fail("Unexpected Exception thrown : " + e.getMessage());
+		}		
+		
+	
+		try {	
+			diagBean.resetDumpOptions();
+		} catch (DumpConfigurationUnavailableException e) {
+			Assert.fail("Unexpected DumpConfigurationUnavailableException : " + e.getMessage());
+		}
+
+		List<String> newDumpOptions = new ArrayList<String>();
+		newDumpOptions = getDumpOptions();
+		boolean res = initialDumpOptions.equals(newDumpOptions);
+		
+		Assert.assertTrue(res, "Reset Dump Options failed");
+		
+	}
+
+	
+	/**
+	 * Function to check if the specified file is present in the directory path 
+	 *
+	 * @param dirPath - Directory path where dumps are created 
+	 * @param fileName - Dump file name
+	 * @return a boolean indicating if the file is found or not
+	 */	
+	private static boolean findAndDeleteFile(String dirPath, String fileName) {
+		BufferedReader reader = null;
+		boolean found = false;
+		File dir = new File(dirPath);
+
+        	File[] files = dir.listFiles();
+
+        	for (File f: files) {
```

Comment:
Formatting: space around ':'.

This loop breaks after the first successful delete: that doesn't seem correct. Please explain.

Reply:
We check for the existence of the dump file, if found we don't loop further.